### DISCUSSION TITLE
Embedded png is removed from SVG

### DISCUSSION THREAD

**Discussion by tjandras** [dlemstra/Magick.NET#724] [Help] [answered]
I have an SVG with embedded PNG images in it, like this:

``` <!DOCTYPE svg PUBLIC " -//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"[]>
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <text x="97.5" y="43" fill="rgb(0,0,0)" font-size="24" font-family="monospace" xmlns="http://www.w3.org/2000/svg">Some text</text>
  <text x="97.5" y="67" fill="rgb(0,0,0)" font-size="24" font-family="monospace" xmlns="http://www.w3.org/2000/svg">asd/asd/asd</text>
  <image x="97.5" y="72" width="500" height="500" xlink:href="data: image/png; base64,<data>" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" />
....
```
But when I open it with Magick.NET and save it again like this: (obviously my real use case include some editing before saving again, but simply saving already "corrupts" the SVG)

```
var rs = new MagickReadSettings
{
    Format = MagickFormat.Svg
};
using (MagickImage img = new MagickImage("1.svg", rs))
{
    img.Write("my.svg", MagickFormat.Svg);
}
```
all embedded images disappear. Everything else seems fine. Did I make any mistake, or is it unsupported to have embedded images? Maybe a bug?

I'm using .NET Core on Windows and the latest Magick.NET-Q16-AnyCPU, Version: 7.21.1 Also tried with a bit older version, same thing happens.

Any help is highly appreciated.

Thanks!

PS: I could not attach the SVG file (unsupported file type), but if needed I figure something out.

**Comment by dlemstra** [comment]
You can zip this SVG file and then upload it. I suspect that loading embeded images is not supported but I will need to use your test file to make sure.

**Reply by tjandras** [reply]
[1.zip](https://github.com/dlemstra/Magick.NET/files/5157692/1.zip)

Thanks for the help!

**Comment by dlemstra** [comment, marked as answer]
The problem appears to be that the data should be encoded differently: `data:image/png;base64,` instead of `data:image/png ;base64 ,`. Those spaces are not allowed by the code that parses the attribute.

**Reply by tjandras** [reply]
I see, the sample code above in the original question does have spaces indeed, my bad. But the SVG I attached to your previous comment does not. And that is not working as well.

**Comment by dlemstra** [comment]
![image](https://user-images.githubusercontent.com/10426229/91939634-78bd4000-ecf6-11ea-8a1c-d53b75927d0a.png)

This is what I see in the SVG file inside the zip.

**Reply by tjandras** [reply]
Yes, you are right, my apologies. I have and created so many svgs that I might mixed some up. This solves the issue. Thank you very much!
